src/ripple/core/SQLInterface.h

```diff
@@ -0,0 +1,674 @@
+        SQLDatabase& db,
+        SQLInterface::SQLLedgerInfo& info,
+        beast::Journal const& j,
+        LedgerIndex ledgerFirstIndex,
+        bool ascendSort) = 0;
+
+    virtual bool
+    loadLedgerInfoByIndexLimitedSorted(
+        SQLDatabase_* db,
+        SQLInterface::SQLLedgerInfo& info,
+        beast::Journal const& j,
+        LedgerIndex ledgerFirstIndex,
+        bool ascendSort) = 0;
+
+    virtual bool
+    loadLedgerInfoByHash(
+        SQLDatabase& db,
+        SQLInterface::SQLLedgerInfo& info,
+        beast::Journal const& j,
+        uint256 const& ledgerHash) = 0;
+
+    virtual bool
+    loadLedgerInfoByHash(
+        SQLDatabase_* db,
+        SQLInterface::SQLLedgerInfo& info,
+        beast::Journal const& j,
+        uint256 const& ledgerHash) = 0;
+
+    virtual uint256
+    getHashByIndex(SQLDatabase& db, LedgerIndex ledgerIndex) = 0;
+
+    virtual uint256
+    getHashByIndex(SQLDatabase_* db, LedgerIndex ledgerIndex) = 0;
+
+    virtual bool
+    getHashesByIndex(
+        SQLDatabase& db,
+        beast::Journal const& j,
+        LedgerIndex ledgerIndex,
+        uint256& ledgerHash,
+        uint256& parentHash) = 0;
+
+    virtual bool
+    getHashesByIndex(
+        SQLDatabase_* db,
+        beast::Journal const& j,
+        LedgerIndex ledgerIndex,
+        uint256& ledgerHash,
+        uint256& parentHash) = 0;
+
+    virtual std::map<LedgerIndex, std::pair<uint256, uint256>>
+    getHashesByIndex(
+        SQLDatabase& db,
+        beast::Journal const& j,
+        LedgerIndex minSeq,
+        LedgerIndex maxSeq) = 0;
+
+    virtual void
+    getHashesByIndex(
+        SQLDatabase_* db,
+        beast::Journal const& j,
+        LedgerIndex minSeq,
+        LedgerIndex maxSeq,
+        std::map<LedgerIndex, std::pair<uint256, uint256>>& map) = 0;
+
+    virtual Json::Value
+    loadTxHistory(
+        SQLDatabase& db,
+        Application& app,
+        LedgerIndex startIndex) = 0;
+
+    virtual int
+    loadTxHistory(
+        SQLDatabase_* db,
+        Application& app,
+        Json::Value& txs,
+        LedgerIndex startIndex,
+        int quantity,
+        bool count) = 0;
+
+    virtual NetworkOPs::AccountTxs
+    getAccountTxs(
+        SQLDatabase& db,
+        Application& app,
+        LedgerMaster& ledgerMaster,
+        beast::Journal& j,
+        AccountID const& account,
+        std::int32_t minLedger,
+        std::int32_t maxLedger,
+        bool descending,
+        std::uint32_t offset,
+        int limit,
+        bool bUnlimited) = 0;
+
+    virtual int
+    getAccountTxs(
+        SQLDatabase_* db,
+        Application& app,
+        LedgerMaster& ledgerMaster,
+        beast::Journal& j,
+        AccountID const& account,
+        NetworkOPs::AccountTxs& ret,
+        std::int32_t minLedger,
+        std::int32_t maxLedger,
+        bool descending,
+        std::uint32_t offset,
+        int limit,
+        int limit_used,
+        bool bUnlimited) = 0;
```

Comment:
Some of these functions have a lot of parameters, some of which are the same types. I would consider packing some of these parameters together inside of a struct, which can then be passed to the function.